### PR TITLE
Add new cop Performance/DefineMethod

### PR DESCRIPTION
**Pull request by rrosenblum**

### CODE REVIEW

File: lib/rubocop/cop/performance/define_method.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    module Performance
+      # This cop is used to identify usages of `module_eval` and suggest
+      # changing them to `define_method`.
+      class DefineMethod < Cop
+        MSG = 'Use `define_method` instead of `module_eval`.'
+
+        def on_send(node)
+          _, method, _ = *node
+
+          return unless method == :module_eval
```
Comment (bbatsov) [inline lib/rubocop/cop/performance/define_method.rb]: Won't this yield some false positives in its current form? After all you might use `module_eval` for something different than a method definition.

File: config/enabled.yml [hunk 2]
```diff
@@ -975,6 +975,10 @@ Lint/Void:
 
 ##################### Performance #############################
 
+Performance/DefineMethod:
+  Description: 'Use `define_method` instead of `module_eval`.'
```
Comment (bbatsov) [inline config/enabled.yml]: ... for dynamic method definition.

File: lib/rubocop/cop/performance/define_method.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+module RuboCop
+  module Cop
+    module Performance
+      # This cop is used to identify usages of `module_eval` and suggest
+      # changing them to `define_method`.
+      class DefineMethod < Cop
+        MSG = 'Use `define_method` instead of `module_eval`.'
+
+        def on_send(node)
+          _, method, _ = *node
```
Comment (bbatsov) [inline lib/rubocop/cop/performance/define_method.rb]: Use names like `_something`.

**Comment by rrosenblum** [comment]
I am considering scrapping this cop. I came across [this issue](https://github.com/JuanitoFatas/fast-ruby/issues/22) in the fast-ruby project last night talks about calling the methods that were created with `module_eval` and `define_method`. It appears that it is less preformant to call a method that was created with `define_method` than it is to call a method that was created with `module_eval`.

**Comment by bbatsov** [comment]
Yeah, this is clearly unclear. :-) Generally I prefer using `define_method` because of its semantics, though.

**Pull request closed without merging**